jsesc.js

```diff
@@ -134,7 +137,7 @@ const jsesc = (argument, options) => {
 	}
 
 	if (!isString(argument)) {
-		if (isMap(argument)) {
+		if (!json && isMap(argument)) {
```

Comment:
A potential solution for #57; `Map` and `Set` don't have `toJSON` methods - and even if they did, they probably wouldn't return the same type, so the expectation here is that we never process a `Map` or `Set` in a special way when targetting JSON.